### DISCUSSION TITLE
Control always zero

### DISCUSSION THREAD

**Discussion by xubo92** [do-mpc/do-mpc#174] [Q&A] [answered]
Hi @4flixt

thank you so much for this useful tool!

I'm new to MPC. I was trying to implement a MPC controller for a linear quadrotor system. I use the linear model from here: [model](https://imgur.com/uBYKsTC),  . And I want to use MPC to launch a quadrotor from ground and stabilize at some height (1 meter).

Here is my sample code: [code](https://www.online-python.com/fbieRaUvyM)

I follow the tutorial of do-mpc step by step and they are very clear. But when I run the control loop and tune a set of params, the control input from MPC is always zero. And consequently all the states are not changed. I'm not sure if I'm  doing something wrong.

Could you please take a look at the sample code and give some suggestions?

Thanks a lot!

**Comment by xubo92** [comment, marked as answer]
problem solved. It's not related to do-mpc.

**Comment by 4flixt** [comment]
Hi @xubo92 ! Great to hear that you managed to fix your problem. I meant to answer you sometimes this week but this way its even better. If you encounter issues (that you cannot identify as bugs, yet), please consider using the Discussions tab. 

I will in fact convert this to a discussion. If you think that your issue is still helpful for others, please also consider describing the problem and how you fixed it.

Best 
4flixt
